vernac/vernacentries.ml

```diff
@@ -122,6 +122,9 @@ module DefAttributes = struct
     in
     let scope = scope_of_locality locality discharge deprecated_thing replacement in
     { scope; locality; polymorphic; program; user_warns; canonical_instance; typing_flags; using; reversible; clearbody }
+
+  let def_attributes = Attributes.Notations.return (parse [])
```

Comment:
this is nonsense

Reply:
Is there a way to re-encapsulate the code in `DefAttributes` under the API of `Attributes`?

I guess that what I want is something like:
```
let def_attributes l = ([], DefAttributes.parse l)
val def_attributes : DefAttributes.t Attributes.attribute
```

Reply:
Something like 
~~~ocaml
let def_attributes ~coercion ~discharge =
  let discharge, deprecated_thing, replacement = discharge in
  let clearbody = match discharge with DoDischarge -> clearbody | NoDischarge -> return None in
  (locality ++ user_warns ++ polymorphic ++ program ++
             canonical_instance ++ typing_flags ++ using ++
             reversible ++ clearbody) >>= fun ((((((((locality, user_warns), polymorphic), program),
         canonical_instance), typing_flags), using),
         reversible), clearbody) ->
    let using = Option.map Proof_using.using_from_string using in
    let reversible = Option.default false reversible in
    let () = if Option.has_some clearbody && not (Lib.sections_are_opened())
      then CErrors.user_err Pp.(str "Cannot use attribute clearbody outside sections.")
    in
    let scope = scope_of_locality locality discharge deprecated_thing replacement in
    return { scope; locality; polymorphic; program; user_warns; canonical_instance; typing_flags; using; reversible; clearbody }

let parse ?(coercion=false) ?(discharge=NoDischarge,"","") f =
  Attributes.parse (def_attributes ~coercion ~discharge) f
~~~

Reply:
Great. Adopted.